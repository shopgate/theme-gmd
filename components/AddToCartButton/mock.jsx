/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
