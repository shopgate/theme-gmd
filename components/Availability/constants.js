/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * The availability states of a product.
 */
export const AVAILABILITY_STATE_OK = 'ok';
export const AVAILABILITY_STATE_WARNING = 'warning';
export const AVAILABILITY_STATE_ALERT = 'alert';
