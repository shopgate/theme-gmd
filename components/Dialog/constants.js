/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const DIALOG_TEXT_MESSAGE = 'DIALOG_TEXT_MESSAGE';
export const MODAL_VARIANT_SELECT = 'MODAL_VARIANT_SELECT';
