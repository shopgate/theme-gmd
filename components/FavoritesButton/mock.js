/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

const mockedProduct1 = {
  productId: 'foo',
};

const currentProduct = {
  productId: 'bar',
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateNotOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId],
    },
  },
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
      [currentProduct.productId]: {
        productData: currentProduct,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId, currentProduct.productId],
    },
  },
};

/**
 * Mocked state without data.
 * @type {{product: {currentProduct: {}, favorites: {}}}}
 */
export const mockedStateEmpty = {
  product: {
    currentProduct: {},
    productsById: {},
  },
  favorites: {
    products: {
      ids: [],
    },
  },
};
