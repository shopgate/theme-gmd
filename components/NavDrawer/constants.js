/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const QUICKLINKS_MENU = 'quicklinks';
