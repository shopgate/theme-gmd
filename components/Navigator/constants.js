/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

 /**
  * Action Types.
  */

export const TOGGLE_NAVIGATOR_CART_ICON = 'TOGGLE_NAVIGATOR_CART_ICON';
export const TOGGLE_NAVIGATOR_SEARCH = 'TOGGLE_NAVIGATOR_SEARCH';
export const TOGGLE_NAV_DRAWER = 'TOGGLE_NAV_DRAWER';
export const TOGGLE_PROGRESS_BAR = 'TOGGLE_PROGRESS_BAR';
export const TOGGLE_LOGIN = 'TOGGLE_LOGIN';
export const SET_NAVIGATOR_BACKGROUND = 'SET_NAVIGATOR_BACKGROUND';
export const SET_NAVIGATOR_COLOR = 'SET_NAVIGATOR_COLOR';
export const SET_NAVIGATOR_SEARCH_QUERY = 'SET_NAVIGATOR_SEARCH_QUERY';
export const SET_FILTER_OPENED = 'SET_FILTER_OPENED';
export const SET_FILTER_CLOSED = 'SET_FILTER_CLOSED';
export const SET_FILTER_ATTRIBUTE_OPENED = 'SET_FILTER_ATTRIBUTE_OPENED';
export const SET_FILTER_ATTRIBUTE_CLOSED = 'SET_FILTER_ATTRIBUTE_CLOSED';
export const SET_SEARCH_ENABLED = 'SET_SEARCH_ENABLED';
export const SET_SEARCH_DISABLED = 'SET_SEARCH_DISABLED';
export const SET_ICON_SHADOW_ENABLED = 'SET_ICON_SHADOW_ENABLED';
export const SET_ICON_SHADOW_DISABLED = 'SET_ICON_SHADOW_DISABLED';
export const SET_TITLE_ENABLED = 'SET_TITLE_ENABLED';
export const SET_TITLE_DISABLED = 'SET_TITLE_DISABLED';
export const SET_NAVIGATOR_DISABLED = 'SET_NAVIGATOR_DISABLED';
export const SET_NAVIGATOR_ENABLED = 'SET_NAVIGATOR_ENABLED';
