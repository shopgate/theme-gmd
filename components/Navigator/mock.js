import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const router = {
  routing: false,
  stack: [
    {
      id: '44a34767-2683-4414-9de4-fd736808fdab',
      params: {},
      pathname: '/',
      pattern: '/',
      query: {},
      state: {},
    },
  ],
};

export const defaultState = {
  navigator: {
    backgroundColor: 'red',
    enabled: true,
    filterOpen: false,
    filterAttributeOpen: false,
    loginOpen: false,
    searchActive: false,
    showCartIcon: true,
    showIconShadow: false,
    showProgressBar: true,
    showSearch: true,
    showTitle: true,
    textColor: 'white',
    viewTracking: false,
  },
  search: {
    suggestion: {},
  },
  router,
  view: {
    isLoading: false,
  },
  ui: {
    general: {
      title: null,
    },
  },
};

/**
 * Creates store.
 * @param {Object} props Props which would override state.navigator props.
 * @return {Object}
 */
export const getStore = props => configureStore([thunk])({
  ...defaultState,
  navigator: {
    ...defaultState.navigator,
    ...props,
  },
  search: {
    suggestions: {},
  },
});
