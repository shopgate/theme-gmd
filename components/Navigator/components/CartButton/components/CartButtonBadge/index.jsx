/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { CART_MAX_ITEMS } from 'Pages/Cart/constants';

/**
 * The cart button badge component.
 * Shows the amount of products in the cart.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const CartButtonBadge = (props) => {
  let productCount = `${props.productCount}`;

  if (props.productCount > CART_MAX_ITEMS) {
    productCount = `${CART_MAX_ITEMS}+`;
  }

  return (
    <div className={props.className}>{productCount}</div>
  );
};

CartButtonBadge.propTypes = {
  productCount: PropTypes.number.isRequired,
  className: PropTypes.string,
};

CartButtonBadge.defaultProps = {
  className: '',
};

export default CartButtonBadge;
