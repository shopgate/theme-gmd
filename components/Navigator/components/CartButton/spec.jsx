import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { cartState } from '@shopgate/pwa-common-commerce/cart/mock';
import { CART_MAX_ITEMS } from 'Pages/Cart/constants';
import CartButtonBadge from './components/CartButtonBadge';

const mockedStore = configureStore();
let store;

describe('<CartButton />', () => {
  // eslint-disable-next-line require-jsdoc
  const createComponent = (state) => {
    // eslint-disable-next-line global-require
    const CartButton = require('./index').default;
    store = mockedStore({
      ...state,
      navigator: { showCartIcon: true },
    });
    return mount(<CartButton store={store} />);
  };

  it('should not be visible with prop visible set to false', () => {
    const wrapper = shallow((
      <CartButton cartProductCount={0} visible={false} activeCartRoute={false} />
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CartButtonBadge).render().text()).toBe('0');

    const buttonProps = wrapper.find('button').props();
    expect(buttonProps.style).toEqual({});
  });

  it('should render with cartProductCount beeing set to 1', () => {
    const wrapper = createComponent(cartState);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CartButtonBadge).render().text()).toBe('1');
  });

  it('should show \'99+\' if the cart amount is higher', () => {
    const state = {
      ...cartState,
      cart: {
        ...cartState.cart,
        items: [
          {
            ...cartState.cart.items[0],
            quantity: 115,
          },
        ],
      },
    };

    const wrapper = createComponent(state);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CartButtonBadge).render().text()).toBe(`${CART_MAX_ITEMS}+`);
  });

  it('should transition if it has cart amount', () => {
    const wrapper = createComponent(cartState);
    expect(wrapper).toMatchSnapshot();
    const buttonProps = wrapper.find('button').props();
    expect(buttonProps.style.transitionDelay).toBe('600ms');
    expect(buttonProps.style.minWidth).toBe(56);
    expect(buttonProps.style.transform).toBe('translate3d(0, 0, 0)');
  });

  it('should not delay transition when the cart page is left', () => {
    // eslint-disable-next-line global-require
    const { UnwrappedCartButton } = require('./index');
    const wrapper =
            shallow(<UnwrappedCartButton cartProductCount={5} visible activeCartRoute={false} />);
    expect(wrapper.state().useAnimationDelay).toBe(true);

    // Visit the cart route ...
    wrapper.setProps({ activeCartRoute: true });
    expect(wrapper.state().useAnimationDelay).toBe(true); // We want animation here.

    // Leave the cart route.
    wrapper.setProps({ activeCartRoute: false });
    expect(wrapper.state().useAnimationDelay).toBe(false); // We don't want animation here!});
  });

  it('should navigate on click', () => {
    const wrapper = createComponent(cartState);
    wrapper.find('button').simulate('click', { preventDefault: () => {} });

    const result = [{
      type: 'NAVIGATE',
      action: 'PUSH',
      location: '/cart',
      state: undefined,
    }];
    expect(store.getActions()).toEqual(result);
  });
});
