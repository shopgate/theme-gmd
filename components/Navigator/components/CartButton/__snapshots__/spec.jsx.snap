// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartButton /> should not be visible with prop visible set to false 1`] = `
<Connect(CartButton)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartButton
    activeCartRoute={true}
    cartProductCount={0}
    openCart={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    visible={false}
  >
    <button
      className="css-nmjls1"
      data-test-id="CartButton"
      onClick={[Function]}
      style={Object {}}
    >
      <Ripple
        className="css-167cazg"
        color="#000"
        disabled={false}
        fill={true}
        onClick={[Function]}
        onComplete={[Function]}
        overflow={false}
        size={null}
      >
        <div
          aria-hidden={true}
          className="css-167cazg"
          data-test-id="Ripple"
          onClick={[Function]}
          style={null}
        >
          <CartIcon>
            <Icon
              className=""
              color={null}
              content="<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>"
              size="inherit"
              viewBox="0 0 24 24"
            >
              <svg
                className="css-yys9hb "
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>",
                  }
                }
                style={
                  Object {
                    "fill": null,
                    "fontSize": "inherit",
                  }
                }
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </Icon>
          </CartIcon>
          <CartButtonBadge
            productCount={0}
          >
            <div
              className={
                Object {
                  "data-css-8teden": "",
                }
              }
            >
              0
            </div>
          </CartButtonBadge>
        </div>
      </Ripple>
    </button>
  </CartButton>
</Connect(CartButton)>
`;

exports[`<CartButton /> should render with cartProductCount beeing set to 1 1`] = `
<Connect(CartButton)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartButton
    activeCartRoute={true}
    cartProductCount={1}
    openCart={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    visible={true}
  >
    <button
      className="css-nmjls1"
      data-test-id="CartButton"
      onClick={[Function]}
      style={
        Object {
          "minWidth": 56,
          "transform": "translate3d(0, 0, 0)",
          "transitionDelay": "600ms",
        }
      }
    >
      <Ripple
        className="css-167cazg"
        color="#000"
        disabled={false}
        fill={true}
        onClick={[Function]}
        onComplete={[Function]}
        overflow={false}
        size={null}
      >
        <div
          aria-hidden={true}
          className="css-167cazg"
          data-test-id="Ripple"
          onClick={[Function]}
          style={null}
        >
          <CartIcon>
            <Icon
              className=""
              color={null}
              content="<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>"
              size="inherit"
              viewBox="0 0 24 24"
            >
              <svg
                className="css-yys9hb "
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>",
                  }
                }
                style={
                  Object {
                    "fill": null,
                    "fontSize": "inherit",
                  }
                }
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </Icon>
          </CartIcon>
          <CartButtonBadge
            productCount={1}
          >
            <div
              className={
                Object {
                  "data-css-8teden": "",
                }
              }
            >
              1
            </div>
          </CartButtonBadge>
        </div>
      </Ripple>
    </button>
  </CartButton>
</Connect(CartButton)>
`;

exports[`<CartButton /> should show '99+' if the cart amount is higher 1`] = `
<Connect(CartButton)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartButton
    activeCartRoute={true}
    cartProductCount={115}
    openCart={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    visible={true}
  >
    <button
      className="css-nmjls1"
      data-test-id="CartButton"
      onClick={[Function]}
      style={
        Object {
          "minWidth": 56,
          "transform": "translate3d(0, 0, 0)",
          "transitionDelay": "600ms",
        }
      }
    >
      <Ripple
        className="css-167cazg"
        color="#000"
        disabled={false}
        fill={true}
        onClick={[Function]}
        onComplete={[Function]}
        overflow={false}
        size={null}
      >
        <div
          aria-hidden={true}
          className="css-167cazg"
          data-test-id="Ripple"
          onClick={[Function]}
          style={null}
        >
          <CartIcon>
            <Icon
              className=""
              color={null}
              content="<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>"
              size="inherit"
              viewBox="0 0 24 24"
            >
              <svg
                className="css-yys9hb "
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>",
                  }
                }
                style={
                  Object {
                    "fill": null,
                    "fontSize": "inherit",
                  }
                }
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </Icon>
          </CartIcon>
          <CartButtonBadge
            productCount={115}
          >
            <div
              className={
                Object {
                  "data-css-8teden": "",
                }
              }
            >
              99+
            </div>
          </CartButtonBadge>
        </div>
      </Ripple>
    </button>
  </CartButton>
</Connect(CartButton)>
`;

exports[`<CartButton /> should transition if it has cart amount 1`] = `
<Connect(CartButton)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartButton
    activeCartRoute={true}
    cartProductCount={1}
    openCart={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    visible={true}
  >
    <button
      className="css-nmjls1"
      data-test-id="CartButton"
      onClick={[Function]}
      style={
        Object {
          "minWidth": 56,
          "transform": "translate3d(0, 0, 0)",
          "transitionDelay": "600ms",
        }
      }
    >
      <Ripple
        className="css-167cazg"
        color="#000"
        disabled={false}
        fill={true}
        onClick={[Function]}
        onComplete={[Function]}
        overflow={false}
        size={null}
      >
        <div
          aria-hidden={true}
          className="css-167cazg"
          data-test-id="Ripple"
          onClick={[Function]}
          style={null}
        >
          <CartIcon>
            <Icon
              className=""
              color={null}
              content="<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>"
              size="inherit"
              viewBox="0 0 24 24"
            >
              <svg
                className="css-yys9hb "
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>",
                  }
                }
                style={
                  Object {
                    "fill": null,
                    "fontSize": "inherit",
                  }
                }
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </Icon>
          </CartIcon>
          <CartButtonBadge
            productCount={1}
          >
            <div
              className={
                Object {
                  "data-css-8teden": "",
                }
              }
            >
              1
            </div>
          </CartButtonBadge>
        </div>
      </Ripple>
    </button>
  </CartButton>
</Connect(CartButton)>
`;
